### PR TITLE
fix: update model calling for examples

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

We have updated the way to set model for agent, but didn't update the content in examples folder

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by zechengz — APPROVED** [review]

File: examples/function_call/role_playing_with_functions.py [hunk 1]
```diff
@@ -60,15 +60,15 @@ def main(model_type=ModelType.GPT_3_5_TURBO, chat_turn_limit=10) -> None:
...
             tools=function_list,
         ),
         user_agent_kwargs=dict(
             model=ModelFactory.create(
                 model_platform=ModelPlatformType.OPENAI,
-                model_type=ModelType.GPT_3_5_TURBO,
+                model_type=model_type,
                 model_config_dict=user_model_config.__dict__,
             ),
         ),
```
Comment (Appointat) [inline examples/function_call/role_playing_with_functions.py:74]: There will be a minor bug if the input parameter `model_type` is not an OpenAI-based model. We need to ensure that `model_platform` and `model_type` match.

File: examples/misalignment/role_playing_multiprocess.py [hunk 2]
```diff
@@ -43,9 +42,7 @@ def generate_data(
         with_task_specify=True,
         with_task_planner=False,
         task_type=TaskType.MISALIGNMENT,
-        task_specify_agent_kwargs=dict(
-            model_config=ChatGPTConfig(temperature=1.4)
-        ),
+        task_specify_agent_kwargs=dict(),
```
Comment (Appointat) [inline examples/misalignment/role_playing_multiprocess.py:45]: The original initial value is `temperature=1.4`, don't we need it anymore?

**Review by Appointat — CHANGES_REQUESTED** [review]
I have reviewed the PR, good job. And do we have added a test for the creation of the single agent? Thanks.

File: examples/misalignment/role_playing_multiprocess.py (hunk 2 above)
Comment (Wendong-Fan) [reply examples/misalignment/role_playing_multiprocess.py:45]: I think it's not important or necessary but nice to have, have added it back

**Comment by Wendong-Fan** [comment]
> I have reviewed the PR, good job. And do we have added a test for the creation of the single agent? Thanks.

We have low test coverage for examples and I agree that we can add more test for examples, but let's focus on core module development first

**Review by Appointat — COMMENTED** [review]
Looks good to me

**Pull request merged**